File: HRMS/HRMS.Web/Modules/ServerTypes/Communication/SocialPostMediaForm.ts
import { EnumEditor, ImageUploadEditor, initFormType, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { SocialMediaType } from "./SocialMediaType";

export interface SocialPostMediaForm {
    FileName: ImageUploadEditor;
    MediaType: EnumEditor;
    DisplayOrder: IntegerEditor;
}

export class SocialPostMediaForm extends PrefixedContext {
    static readonly formKey = 'Communication.SocialPostMedia';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SocialPostMediaForm.init) {
            SocialPostMediaForm.init = true;

            var w0 = ImageUploadEditor;
            var w1 = EnumEditor;
            var w2 = IntegerEditor;

            initFormType(SocialPostMediaForm, [
                'FileName', w0,
                'MediaType', w1,
                'DisplayOrder', w2
            ]);
        }
    }
}

[SocialMediaType]; // referenced types

File: HRMS/HRMS.Web/Modules/Communication/SocialPost/SocialPostMediaForm.cs
using Serenity.ComponentModel;

namespace HRMS.Communication.Forms;

[FormScript("Communication.SocialPostMedia")]
[BasedOnRow(typeof(SocialPostMediaRow), CheckNames = true)]
public class SocialPostMediaForm
{
    public string FileName { get; set; }
    public SocialMediaType MediaType { get; set; }
    public int DisplayOrder { get; set; }
}
